libs/algo/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+use pyo3::prelude::{pymodule, PyModule, PyResult, Python};
+use numpy::{PyArray4, PyArray3, ToPyArray};
+use ndarray::{array, Array, ArrayBase, ArrayView, ArrayViewMut, Axis, Ix1, Ix2, Ix3, Ix4};
+
+
+fn geomedian<'a>(in_array: ArrayView<'a, f32, Ix4>, max_iter: usize, eps: f32) -> Array<f32, Ix3> {
```

Comment:
What's the situation with GIL? Is it released, when is it released, when is it re-acquired?

Reply:
atm the GIL is just held the entire time, but I think I'll end up releasing it for the call to `geomedian` - what do you reckon?

Reply:
Has to be once per top level call to be fast enough